plugins/pcap/main.cpp

```diff
@@ -661,6 +661,15 @@ class writer : public format::writer {
   using format::writer::write;
 
   caf::error write(const table_slice& slice) override {
+    auto&& layout = slice.layout();
+    // TODO: relax this check. We really only need the (1) flow, and (2) PCAP
+    // payload. Everything else is optional.
+    if (!congruent(layout, make_packet_type())) {
+      return caf::make_error(
+        ec::format_error, fmt::format("pcap-writer is unable to write batch "
+                                      "of schema '{}': expected 'pcap.packet'",
+                                      slice.layout()));
+    }
```

Comment:
I would do this not via congruency, now that the writer gets everything. Two approaches come to mind:

1. Just look at the schema name and bail out if it's not `pcap.packet`
2. Ensure that the blob payload has the right attribute value, e.g., `libpcap-packet`, to ensure that you can stuff it into the libpcap function without producing garbage

Reply:
I did not change this logic, I just moved it up a bit. I would prefer not to touch this for now as it's not currently something I want to invest time in.

Reply:
But you fundamentally changed what data goes into reader, so you must adapt the logic accordingly. You cannot see this in isolation, the effect of your change is across multiple components that now must be re-harmonized.

Reply:
In other words: this wasn't an issue in the past, but now it is.

Reply:
I chose to restore the type-name restriction